src/consumption/dtos/sunbird/project.js

```diff
@@ -0,0 +1,151 @@
+'use strict'
+const utils = require('@generics/utils')
+const common = require('@constants/common')
+
+class ProjectTemplateDTO {
+	/**
+	 * Public entry — accepts a single template object or an array of them.
+	 * Mirrors the pattern used in UserDTO.transform from your example.
+	 */
+	static async transform(input = {}) {
+		// If it's an array of templates
+		if (Array.isArray(input)) {
+			return Promise.all(input.map((t) => ProjectTemplateDTO.transformTemplate(t)))
+		}
+
+		// If it's a single template
+		return ProjectTemplateDTO.transformTemplate(input)
+	}
+
+	/**
+	 * Core conversion logic — preserved from your original function.
+	 */
+	static async transformTemplate(templateData = {}) {
+		try {
+			const concepts =
+				templateData?.concepts && Array.isArray(templateData?.concepts)
+					? templateData?.concepts
+					: templateData?.concepts && typeof templateData?.concepts == common.STRING
+					? templateData?.concepts.split(',').map((c) => c.trim())
+					: [] || []
+
+			let template = {
+				title: templateData.title,
+				description: templateData.objective || '',
+				concepts,
+				keywords: utils.formatKeywords(templateData.keywords),
+				isDeleted: false,
+				createdBy: templateData.user_id,
+				updatedBy: templateData.user_id,
+				learningResources: utils.convertResources(templateData.learning_resources || []),
+				isReusable: true,
+				deleted: false,
+				status: common.PUBLISHED_STATUS,
+				externalId: utils.generateExternalId(templateData.title),
+				entityType: '',
+				metaInformation: utils.formatProjectMetaInformation(templateData),
+				averageRating: 5,
+				noOfRatings: 1,
+				ratings: {
+					1: 0,
+					2: 0,
+					3: 0,
+					4: 0,
+					5: 1,
+				},
+				taskCreationForm: '',
+				recommendedFor: [], //Initially empty
+				categories: [], //Initially empty
+				tasks: [], // Initially empty
+				taskSequence: [], // Initially empty
+				createdAt: new Date(),
+				updatedAt: new Date(),
+				__v: 0,
+			}
+			return { success: true, template }
+		} catch (error) {
+			console.error('Error in formatTemplate:', error.message)
+			return { success: false, error: error.message }
+		}
+	}
+}
+
+/**
+ * UserProgramMapping DTO
+ * Normalizes a user-program mapping document into a predictable JS object.
+ * Exported as `userProgramMapping` on the default export for backwards compatibility.
+ */
+class UserProgramMappingDTO {
```

Comment:
@adithyadinesh0412 Can we move the business logic to helper ? DTO we can use for data transformation only